examples/import-sample-export-gnn.ipynb

```diff
@@ -0,0 +1,350 @@
+    "y = torch.tensor(sample_node_properties[\"subject\"], dtype=torch.long)\n",
+    "\n",
+    "data = Data(x=x, y=y, edge_index=edge_index)\n",
+    "num_classes = y.unique().shape[0]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b1885ce",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Do a random split of the data so that ~10% goes into a test set and the rest used for training\n",
+    "transform = RandomNodeSplit(num_test=40, num_val=0)\n",
+    "transform(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4f9ceb11",
+   "metadata": {},
+   "source": [
+    "## Training and evaluating a GCN\n",
+    "\n",
+    "Let's now define and traing the GCN using PyG and our sampled CORA as input. We adapt the CORA GCN example from the [PyG documentation](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html#learning-methods-on-graphs).\n",
+    "\n",
+    "In this example we evaluate the model on a test set of the sampled CORA. Please not however, that since GCN is an inductive algorithm we could also have evaluated it on the full CORA dataset, or even another (similar) graph for that matter."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "74b65751",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_classes = y.unique().shape[0]\n",
+    "\n",
+    "# Define the GCN architecture\n",
+    "class GCN(torch.nn.Module):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "        self.conv1 = GCNConv(data.num_node_features, 16)\n",
+    "        self.conv2 = GCNConv(16, num_classes)\n",
+    "\n",
+    "    def forward(self, data):\n",
+    "        x, edge_index = data.x, data.edge_index\n",
+    "\n",
+    "        x = self.conv1(x, edge_index)\n",
+    "        x = F.relu(x)\n",
+    "        x = F.dropout(x, training=self.training)\n",
+    "        x = self.conv2(x, edge_index)\n",
+    "\n",
+    "        return F.log_softmax(x, dim=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "02e8fadb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "device = torch.device(\"cuda\" if torch.cuda.is_available() else \"cpu\")\n",
+    "model = GCN().to(device)\n",
+    "data = data.to(device)\n",
+    "optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)\n",
+    "\n",
+    "# Train the GCN using the CORA sample represented by `data`\n",
+    "model.train()\n",
+    "for epoch in range(200):\n",
+    "    optimizer.zero_grad()\n",
+    "    out = model(data)\n",
+    "    loss = F.nll_loss(out[data.train_mask], data.y[data.train_mask])\n",
+    "    loss.backward()\n",
+    "    optimizer.step()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9169bf30",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Evaluate the trained GCN model on our test set\n",
+    "model.eval()\n",
```

Comment:
does this also produce some metrics? i dont understand how that would work because the model doesnt know the data nor the test_mask right

Reply:
https://stackoverflow.com/questions/60018578/what-does-model-eval-do-in-pytorch

Reply:
As is mentioned in the notebook that code is taken directly from the PyG example https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html#learning-methods-on-graphs

Reply:
thanks! that makes sense